### PR TITLE
ISPN-11813 Built-in server configuration

### PR DESCRIPTION
**Pull request by tristantarrant**
https://issues.redhat.com/browse/ISPN-11813

### CODE REVIEW

File: server/runtime/src/main/resources/infinispan-server-internal.xml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+         <interface name="public">
+            <inet-address value="${infinispan.bind.address:127.0.0.1}"/>
+         </interface>
+      </interfaces>
+
+      <socket-bindings default-interface="public" port-offset="${infinispan.socket.binding.port-offset:0}">
+         <socket-binding name="default" port="${infinispan.bind.port:11222}"/>
+      </socket-bindings>
+
+      <endpoints socket-binding="default">
```
Comment (karesti) [inline server/runtime/src/main/resources/infinispan-server-internal.xml:23]: security is not enabled ?

**Comment by karesti** [comment]
@tristantarrant I personally don't like the '-'
I prefer a wold but maybe this has been chosen for historical reasons ?

**Comment by tristantarrant** [comment]
You are right Katia, but I didn't want to introduce a new config option because that would require checking for conflicts with the `-c` option.
My other proposal then would be `-i` (short) `--builtin` (long) and make sure the user doesn't also specify `-c`.
As for security, I feel that such a configuration would only be demo-worthy and not usable for development/production, so I've made it as simple as possible.

**Comment by karesti** [comment]
@tristantarrant I prefer -i and --builtIn rather than '-' and the proper documentation of this option

**Comment by tristantarrant** [comment]
Honestly, I'm thinking this should probably be a hidden option :)

**Comment by ryanemerson** [comment]
:-1: for me. I don't think we should be adding a feature/flag in production to avoid having to provide a `.xml` configuration file when testing.

**Comment by wburns** [comment]
I would say we can just close this for now. I can live without it. Feel free to reopen if you want @tristantarrant

**Pull request closed without merging**